cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -199,6 +203,24 @@ impl LockedManifest {
         let contents = fs::read(path).map_err(LockedManifestError::ReadLockfile)?;
         serde_json::from_slice(&contents).map_err(LockedManifestError::ParseLockfile)
     }
+
+    pub fn check_lockfile(
+        path: &CanonicalPath,
```

Comment:
note nonblocking: I don't see much reason to do this in C++. Reading the lockfile is within the bounds of something Rust is allowed to do. Generating a warning to print to users is not really within the bounds of what pkgdb should be doing. And we have better context on the Rust side anyways (was this package just installed, or was it already installed). Exec'ing to diff some JSON doesn't seem worth it

Reply:
(nonblocking)
I agree that verification of validity of the lockfile belongs on the pkgdb side

I don't think the contents of the lockfile should be opaque on the Rust side, which is why I think things like https://github.com/flox/flox/blob/fa1234d5a475d2a292303a90cd37abbebdee1c44/cli/flox/src/commands/environment.rs#L2090 are okay

Printing a warning to a user based on the contents of a lockfile is different than verifying that the lockfile is valid

Reply:
Yes, rust can read the lcokfile, and if we _always_ warn even if `allow.* = true`, even my concern about interpreting doe not hold.
Then again since we already have this logic in cpp for validation, I wouldn't implement _the same_ logic again in rust.

We can come back to this, but since you marked this as nonblocking twice I'm going to resolve for now